pyEpiabm/pyEpiabm/tests/test_sweep/test_host_progression_sweep.py

```diff
@@ -176,6 +189,116 @@ def test_icdf_exception_raise(self):
+            real = pe.Parameters.instance().time_steps_per_day
+
+            # Assigns temporarily a new value for time steps per day to test
+            # multiple time steps per day case
+            pe.Parameters.instance().time_steps_per_day = 100
+            # Same test as test_infectiousness_progression:
+            infectious_period = pe.Parameters.instance().asympt_infect_period
+            model_time_step = 1 / pe.Parameters.instance().time_steps_per_day
+            num_infectious_ts = int(np.ceil(infectious_period /
+                                            model_time_step))
+            test_sweep = pe.sweep.HostProgressionSweep()
+            infect_prog = test_sweep.infectiousness_progression
+            # Checks output type is numpy array
+            self.assertIsInstance(infect_prog, np.ndarray)
+            # Checks elements are 0 after k and greater than 0 before
+            tail = infect_prog[num_infectious_ts:2550]
+            zeros = np.zeros(2550-num_infectious_ts)
+            self.assertTrue((tail == zeros).all())
+            self.assertTrue((infect_prog[0:num_infectious_ts] >
+                            np.zeros(num_infectious_ts)).all())
+
+            # Assigns temporarily a new value for time steps per day to raise
+            # error
+            pe.Parameters.instance().time_steps_per_day = 10000
+            with self.assertRaises(ValueError):
+                pe.sweep.HostProgressionSweep()
+
+            # Resets the value of time steps per day
+            pe.Parameters.instance().time_steps_per_day = real
+        except Exception as e:
+            raise e
+
+    def test_limit_case_infectiousness_progression(self):
+        """Tests the case in the infectiousness progression method where the
+        parameter j would be greater or equal to the infectiousness profile
+        resolution. In that case, the elements of the array infectiousness
+        progression are simply set to 0.
+        """
+        num_infectious_ts =\
+            int(np.ceil(pe.Parameters.instance().asympt_infect_period
+                        * pe.Parameters.instance().time_steps_per_day))
+        with mock.patch('numpy.floor') as mock_floor:
+            # We need to mock the np.floor function to have j greater or equal
+            # to the infectiousness profile resolution
+            mock_floor.return_value = 57
+            test_sweep = pe.sweep.HostProgressionSweep()
+            infect_prog = test_sweep.infectiousness_progression
+            # Checks that the output is a numpy array
+            self.assertIsInstance(infect_prog, np.ndarray)
+            # Checks that all elements are equal to 0
+            zeros = np.zeros(2550)
+            self.assertTrue((infect_prog == zeros).all())
+            # Checks that it is called the right number of times
+            self.assertEqual(mock_floor.call_count, num_infectious_ts + 1)
+
+    def test_updates_infectiousness(self):
```

Comment:
Can we have an explicit test for when there are multiple timesteps per day, to make sure people don't progress too fast (i.e. they still step on once per day not once per time step). And make clear if you also support multiple days per timestep (I think we do so far in the rest of the codebase)

Reply:
See test_infectiousness_progression_small_time_steps

Reply:
i might be misunderstanding, but that test only checks that the array is filled in correctly doesn't it? It would be good to check with the host sweep itself that people there only change infectiousness on full days - its possible otherwise a person could step through the array at the wrong rate (I know this implementation works but I think its something that could go wrong very easily if we ever have to refactor here for any reason)

Reply:
I'm not sure I understand, why would we want the infectiousness to only be updated on full days if we have smaller time steps? Maybe there is a functionality that I don't understand in the code, this is not what I understood from my reading of CovidSim. 

Yes the test only checks that the infectiousness progression array is filled in correctly (this was the goal of the test). Are you suggesting that in test_updates_infectiousness there should be a subtest with the case where the time step is smaller than 1?